program/src/instructions/shared_accounts.rs

```diff
@@ -0,0 +1,235 @@
+    #[account(address = escrow::ID)]
+    pub escrow_program: Option<UncheckedAccount<'info>>,
+}
+
+/* Shared account structs for different standards */
+
+/// Shared accounts for interacting with Metaplex legacy and pNFTs.
+#[derive(Accounts)]
+pub struct MplxShared<'info> {
+    /// The Token Metadata metadata account of the NFT.
+    /// CHECK: ownership, structure and mint are checked in assert_decode_metadata.
+    #[account(mut)]
+    pub metadata: UncheckedAccount<'info>,
+
+    // /// Either the legacy token program or token-2022.
+    // pub token_program: Interface<'info, TokenInterface>,
+    // /// The SPL associated token program.
+    // pub associated_token_program: Program<'info, AssociatedToken>,
+    // /// The Solana system program.
+    // pub system_program: Program<'info, System>,
+
+    // --------------------------------------- pNft
+    /// The Token Metadata edition account of the NFT.
+    /// CHECK: seeds checked on Token Metadata CPI
+    //note that MASTER EDITION and EDITION share the same seeds, and so it's valid to check them here
+    pub edition: UncheckedAccount<'info>,
+
+    // Todo: add ProgNftShared back in, if possible
+    // pub pnft_shared: ProgNftShared<'info>,
+    /// The Token Metadata program account.
+    /// CHECK: address constraint is checked here
+    #[account(address = mpl_token_metadata::ID)]
+    pub token_metadata_program: Option<UncheckedAccount<'info>>,
+
+    /// The sysvar instructions account.
+    /// CHECK: address constraint is checked here
+    #[account(address = anchor_lang::solana_program::sysvar::instructions::ID)]
+    pub sysvar_instructions: Option<UncheckedAccount<'info>>,
+
+    /// The Metaplex Token Authority Rules account that stores royalty enforcement rules.
+    /// CHECK: validated by mplex's pnft code
+    pub authorization_rules: Option<UncheckedAccount<'info>>,
+
+    /// The Metaplex Token Authority Rules program account.
+    /// CHECK: address constraint is checked here
+    #[account(address = MPL_TOKEN_AUTH_RULES_ID)]
+    pub authorization_rules_program: Option<UncheckedAccount<'info>>,
+}
+
+/// Additional accounts for Metaplex legacy & pNFT trade instructions
+#[derive(Accounts)]
+pub struct MplxTradeShared<'info> {
+    /// The Token Metadata source token record account of the NFT.
+    /// CHECK: seeds checked on Token Metadata CPI
+    #[account(mut)]
+    pub taker_token_record: Option<UncheckedAccount<'info>>,
+
+    /// The Token Metadata token record for the pool.
+    /// CHECK: seeds checked on Token Metadata CPI
+    #[account(mut)]
+    pub pool_token_record: Option<UncheckedAccount<'info>>,
+}
+
+/// Additional accounts for Metaplex legacy & pNFT transfer instructions
+#[derive(Accounts)]
+pub struct MplxTransferShared<'info> {
+    /// The Token Metadata source token record account of the NFT.
+    /// CHECK: seeds checked on Token Metadata CPI
+    #[account(mut)]
+    pub owner_token_record: Option<UncheckedAccount<'info>>,
```

Comment:
What if we just called it `user_token_record` for both Trade + Transfer and only have `MplxShared`?

Reply:
Might hose the resolvers; let me see if I can make it work.

Reply:
This might hose the resolvers but I'll see if I can make it work. We might have to rename owner & taker --> user then it would be consistent.

Reply:
Works 👍